cmd/snap-confine/mount-support.c

```diff
@@ -625,6 +625,7 @@ void sc_populate_mount_ns(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 			{"/sys"},	// fundamental filesystem
 			{"/tmp"},	// to get writable tmp
 			{"/var/snap"},	// to get access to global snap data
+			{"/var/lib/dhcp",.is_optional = true}, // to support network-control interface
```

Comment:
I think this may be a bit too interface specific. Perhaps a better place would be to extend the network control interface and add it in `MountConnectedPlug()` like we do for desktop fonts. WDYT?

Reply:
I'm exploring this alternative.

Reply:
The alternative is now proposed as https://github.com/snapcore/snapd/pull/8123

Reply:
Re this being too interface-specific, I'll counter and saw that there a bunch of things in this list that are already interface specific:
* /usr/src (system-trace, marked as FIXME)
* /var/log (log-observe, marked as FIXME)
* /run/media, /media (removable-media)
* /run/netns (network-control)
* /mnt (removable-media)

Of course, /home, /root, /run, /etc are all interface specific in one way or another. In that light, adding one more (with a FIXME comment like with /var/log and /usr/src) does not seem unjustified for a simple bug fix.

The question then becomes, do we want to revamp a bunch of this for a bug fix or do we want to add the one line and do the revamp in a second step?

Reply:
The revamp is in the other PR :) I think the other PR is better way to handle this but requires some more polish.